exploratory_data_analysis/pubtator_exporatory_data_analysis.py

```diff
@@ -0,0 +1,257 @@
+                        "section": section_headers
+                        if len(section_headers) > 0
+                        else section_headers_type,
+                        "published_year": year[0]
+                        if len(year) > 0
+                        else 0,  # Shouldnt get to the else statement
+                    }
+                )
+
+
+# # Analyze Abstract/Full Text Dataset
+
+pubtator_central_metadata_df = pd.read_csv("output/pmc_metadata.tsv.xz", sep="\t")
+print(pubtator_central_metadata_df.shape)
+pubtator_central_metadata_df.head()
+
+# ## Sanity Check the data
+
+# Sanity check that all documents have a published year
+(pubtator_central_metadata_df.query("published_year.isnull()").shape)
+
+# Do all documents have an id?
+(pubtator_central_metadata_df.query("doc_id.isnull()").shape)
+
+# Do all documents have a pmid?
+(pubtator_central_metadata_df.query("pmid.isnull()").shape)
+
+# Do all documents have a pmc id?
+(pubtator_central_metadata_df.query("pmc.isnull()").shape)
+
+# Do all documents have a doi?
+(pubtator_central_metadata_df.query("doi.isnull()").shape)
+
+# ## Published Year Distribution
+
+(pubtator_central_metadata_df.sort_values("published_year").published_year.unique())
+
+doc_count_df = (
+    pubtator_central_metadata_df.groupby("published_year")
+    .agg({"published_year": "size"})
+    .rename(index=str, columns={"published_year": "doc_count"})
+    .reset_index()
+    .astype({"published_year": int, "doc_count": int})
+)
+doc_count_df.head()
+
+g = (
+    p9.ggplot(
+        doc_count_df.query("published_year > 0& published_year < 1950"),
+        p9.aes(x="published_year", y="doc_count"),
+    )
+    + p9.geom_col(position=p9.position_dodge(width=0.9), fill="#1f78b4")
+    + p9.labs(
+        title="Number of Abstracts Pre 1950", x="Publication Year", y="Document Count"
+    )
+)
+g.save("output/figures/pre_1950_doc_count.png", dpi=500)
+print(g)
+
+g = (
+    p9.ggplot(
+        doc_count_df.query("published_year >= 1950"),
+        p9.aes(x="published_year", y="doc_count"),
+    )
+    + p9.geom_col(position=p9.position_dodge(width=0.9), fill="#1f78b4")
+    + p9.labs(
+        title="Number of Abstracts Post 1950", x="Publication Year", y="Document Count"
+    )
+)
+g.save("output/figures/post_1950_doc_count.png", dpi=500)
+print(g)
+
+# # Shared Tokens Across Time
+
+tokens_by_year = defaultdict(Counter)
+sentence_iterator = PubMedSentencesIterator(
+    pubtator_abstract_batch, year_filter=list(range(1990, 2022, 1)), return_year=True
+)
+
+if not Path("output/unique_tokens_by_year.pkl").exists():
+    for year, sentence in tqdm.tqdm(sentence_iterator):
+        tokens_by_year[year].update(Counter(sentence))
+
+if not Path("output/unique_tokens_by_year.pkl").exists():
+    pickle.dump(tokens_by_year, open("output/unique_tokens_by_year.pkl", "wb"))
+else:
+    tokens_by_year = pickle.load(open("output/unique_tokens_by_year.pkl", "rb"))
+
+# ## Unique Tokens Available per Year
```

Comment:
Does the increase in unique tokens mean that people are coming up with more concepts and new words to describe them? Or is this just a result of seeing rarer words because there are more publications?

Reply:
>Does the increase in unique tokens mean that people are coming up with more concepts and new words to describe them? Or is this just a result of seeing rarer words because there are more publications?

The latter is more likely as an increase in unique tokens is positively correlated with number of documents being made available via open access licenses. If I had access to all of the old documents, I hypothesize that there wouldn't be this strong of a discrepancy.